frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/utils.ts

```diff
@@ -68,6 +74,7 @@ export const getPipelineRunData = (
     {},
     pipeline?.metadata?.annotations,
     latestRun?.metadata?.annotations,
+    !latestRun?.metadata?.annotations?.preferredName && { preferredName: pipelineName },
```

Comment:
Only real question here is do we want to call this something else.

cc @vdemeester 

Essentially what's happening here is we have a PipelineRun without a reference to a Pipeline. However, we build our PipelineRun names off of the Pipeline name + 6 random characters. Without that, we sorta need some clean reference otherwise we'll end up with PipelineRun + 6 random characters + 6 random... etc. So we are going with the first one we set in the annotations, "preferredName" was off the top of my head. Thoughts?

Reply:
Do we want to *namespace* the annotation name, like `io.openshift.pipeline/preferredName` ?
Otherwise, seems good to me.

What happens if the pipelinerun doesn't have this annotation ? Will it then add the annotation for the next one ?

Reply:
Thanks for the input @vdemeester 
yes, if the pipeline run doesn't have this annotation, it will be added to the next one.
I think we could have the annotation namespaced.  cc @andrewballantyne

Reply:
Yup. Namespacing it sounds like a good idea